test/test.oauth2.ts

```diff
@@ -1149,6 +1150,36 @@ describe('oauth2', () => {
           done();
         });
       });
+
+      it('should call refreshHandler and set credential if token expires in request()', async () => {
+        const scope = nock('http://example.com')
+          .get('/access')
+          .reply(code, {
+            error: {code, message: 'Invalid Credentials'},
+          });
+        const expectedRefreshedAccessToken = {
+          access_token: 'access_token',
+          expiry_date: 123456789,
+        };
+        client.refreshHandlerCallback = async () => {
+          return expectedRefreshedAccessToken;
+        };
+        client.setCredentials({
+          access_token: 'initial-access-token',
+          expiry_date: new Date().getTime() - 1000,
+        });
+        client.request({url: 'http://example.com'}, () => {
```

Comment:
This also returns a Promise. I suggest just testing the promise response.
You also need to assert that the authorization header is added.

Reply:
Honestly, I think the idea here is only checking for accessTokens being refreshed and there're enough test cases for requestHeaders already, I added an assert checking authorization header is being added.

In this file, there were other 19 client.request() being used without testing promise. I am not sure it is consistent with current code-style to declare a new responseType(for test purpose) and only testing two of client.request(). If @bcoe is okay with this, I'll test on the Promise.